src/components/screensavermanager.js

```diff
@@ -18,6 +20,42 @@ define(["events", "playbackManager", "pluginManager"], function (events, playbac
         }
     });
 
+    function getScreensaverPlugin() {
+
+        var apiClient = connectionManager.currentApiClient();
+
+        if (apiClient && apiClient.isLoggedIn()) {
+            isLoggedIn = true;
+        }
+        var option = function () {
+            try {
+                return userSettings.get("screensaver", false);
+            } catch (err) {
+                return null;
+            }
+        }();
+
+        var defaultOption = function (isLoggedIn) {
+            if (isLoggedIn) {
+                return "backdropscreensaver";
+            }
+
+            return "logoscreensaver";
+        }(isLoggedIn);
+
+        if ((option = option || defaultOption) === "none") {
+            return null;
+        }
+
+        var plugin = pluginManager.ofType("screensaver").filter(function (i) {
+            return i.id === option;
+        })[0];
+
+        return plugin = plugin || pluginManager.ofType("screensaver").filter(function (i) {
+            return i.id === defaultOption;
+        })[0];
```

Comment:
eh, this looks like uglified code we fought before :) please don't, write slightly more verbose code, no one is going to object